test/smoke/src/areas/positron/welcome/welcome.test.ts

```diff
@@ -0,0 +1,184 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2024 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+
+import { expect } from '@playwright/test';
+import { Application, Logger, PositronPythonFixtures, PositronRFixtures } from '../../../../../automation';
+import { installAllHandlers } from '../../../utils';
+
+/*
+ *  Welcome test cases.
+ */
+
+export function setup(logger: Logger) {
+	describe('Welcome Page', () => {
+		installAllHandlers(logger);
+
+		beforeEach(async function () {
+			const app = this.app as Application;
+
+			await app.workbench.quickaccess.runCommand('Help: Welcome');
+			app.workbench.editors.waitForActiveEditor('Welcome');
+		});
+
+		afterEach(async function () {
+			const app = this.app as Application;
+			await app.workbench.quickaccess.runCommand('View: Close All Editors');
+		});
+
+		it('Verify Welcome page header and footer', async function () {
+			const app = this.app as Application;
+
+			await expect(app.workbench.positronWelcome.logoLocator).toBeVisible();
+			const logoBox = await (await app.workbench.positronWelcome.logoLocator.elementHandle())?.boundingBox();
+
+			expect(logoBox?.height).toBeGreaterThan(0);
+			expect(logoBox?.width).toBeGreaterThan(0);
+
+			// product name in release is 'Positron' and in dev is 'Positron Dev'
+			await expect(app.workbench.positronWelcome.titleLocator).toHaveText([/(Positron)|(Positron Dev)/, 'an IDE for data science']);
+
+			await expect(app.workbench.positronWelcome.footerLocator).toHaveText('Show welcome page on startup');
+		});
+
+		it('Verify Welcome page content', async function () {
+			const app = this.app as Application;
+			const OPEN_BUTTONS_COUNT = process.platform === 'darwin' ? 3 : 4;
+			const OPEN_BUTTONS_LABELS = process.platform === 'darwin' ?
+				['Open...', 'New Folder...', 'New Folder from Git...']
+				: ['Open...', 'Open File...', 'New Folder...', 'New Folder from Git...'];
+
+			await expect(app.workbench.positronWelcome.startTitle).toHaveText('Start');
+
+			await expect(app.workbench.positronWelcome.startButtons).toHaveCount(4);
+			await expect(app.workbench.positronWelcome.startButtons).toHaveText(['New Notebook', 'New File', 'New Console', 'New Project']);
```

Comment:
Is there a reason why you would want to know that there are 4 start buttons but that they _didn't_ have the contents of the next assertion? Otherwise, I think we could leave off the assertion of length as its implicit in the `.toHaveText()` assertion.

Reply:
Good point. I'll take it out.